Dijkstra Algorithm/Dijkstra.playground/Contents.swift

```diff
@@ -0,0 +1,137 @@
+//: Playground - noun: a place where people can play
+
+import Foundation
+import UIKit
+
+open class Vertex: Hashable, Equatable {
+
+    open var identifier: String!
+    open var neighbors: [(Vertex, Double)] = []
+    open var pathLengthFromStart: Double = Double.infinity
+    open var pathVerticesFromStart: [Vertex] = []
+
+    public init(identifier: String) {
+        self.identifier = identifier
+    }
+
+    open var hashValue: Int {
+        return self.identifier.hashValue
+    }
+
+    public static func ==(lhs: Vertex, rhs: Vertex) -> Bool {
+        if lhs.hashValue == rhs.hashValue {
+            return true
+        }
+        return false
+    }
+
+    open func clearCache() {
+        self.pathLengthFromStart = Double.infinity
+        self.pathVerticesFromStart = []
+    }
+}
+
+public class Dijkstra {
+    private var totalVertices: Set<Vertex>
+
+    public init(vertices: Set<Vertex>) {
+        self.totalVertices = vertices
+    }
+
+    private func clearCache() {
+        self.totalVertices.forEach { $0.clearCache() }
+    }
+
+    public func findShortestPaths(from startVertex: Vertex) {
+        self.clearCache()
+        startVertex.pathLengthFromStart = 0
+        startVertex.pathVerticesFromStart.append(startVertex)
+        var currentVertex: Vertex! = startVertex
+        while currentVertex != nil {
```

Comment:
This method depending on an IUO (implicitly unwrapped optional) makes me nervous.

For starters, instead of using `Vertex!`, you can have it as `Vertex?` and achieve the same looping logic:

```
while let vertex = currentVertex { ... }
```

Try to refactor this method based on that.